taxonomy-api/taxonomy-actors/src/main/scala/org/sunbird/actors/CategoryInstanceActor.scala

```diff
@@ -0,0 +1,144 @@
+    val frameworkId = request.getRequest.getOrDefault(Constants.FRAMEWORK, "").asInstanceOf[String]
+    val code = request.getRequest.getOrDefault(Constants.CODE, "").asInstanceOf[String]
+    if (frameworkId.isEmpty()) throw new ClientException("ERR_INVALID_FRAMEWORK_ID", s"Invalid FrameworkId: '${frameworkId}' for Categoryinstance ")
+    if (!request.getRequest.containsKey(Constants.CODE)) throw new ClientException("ERR_CATEGORY_CODE_REQUIRED", "Unique code is mandatory for categoryInstance")
+    val getCategoryReq = new Request()
+    getCategoryReq.setContext(new util.HashMap[String, AnyRef]() {{
+        putAll(request.getContext)
+    }})
+    getCategoryReq.getContext.put(Constants.SCHEMA_NAME, Constants.FRAMEWORK_SCHEMA_NAME)
+    getCategoryReq.getContext.put(Constants.VERSION, Constants.FRAMEWORK_SCHEMA_VERSION)
+    getCategoryReq.put(Constants.IDENTIFIER, frameworkId)
+    DataNode.read(getCategoryReq).map(node => {
+      if (null != node && StringUtils.equalsAnyIgnoreCase(node.getIdentifier, frameworkId)) {
+        request.getRequest.put(Constants.IDENTIFIER, CategoryInstanceManager.generateIdentifier(frameworkId, code))
+        DataNode.create(request).map(node => {
+          ResponseHandler.OK.put(Constants.IDENTIFIER, node.getIdentifier)
+            .put(Constants.VERSION_KEY, node.getMetadata.get("versionKey"))
+        })
+      } else throw new ClientException("ERR_INVALID_FRAMEWORK_ID", s"Invalid FrameworkId: '${frameworkId}' for Categoryinstance ")
+    }).flatMap(f => f)
+  }
+
+  private def read(request: Request): Future[Response] = {
+    val requestMethod = request.getRequest.getOrDefault("REQ_METHOD", "").asInstanceOf[String]
+    RequestUtil.restrictProperties(request)
+    val frameworkId = request.getRequest.getOrDefault(Constants.FRAMEWORK, "").asInstanceOf[String]
+    val categoryId = request.getRequest.getOrDefault(Constants.IDENTIFIER, "").asInstanceOf[String]
+    if (frameworkId.isEmpty()) throw new ClientException("ERR_INVALID_FRAMEWORK_ID", s"Invalid FrameworkId: '${frameworkId}' for Categoryinstance ")
+    if (categoryId.isEmpty()) throw new ClientException("ERR_INVALID_CATEGORY_ID", s"Invalid CategoryId: '${categoryId}' for categoryInstance")
+    val categoryInstanceId = CategoryInstanceManager.generateIdentifier(frameworkId, categoryId)
+    val getCategoryReq = new Request()
+    getCategoryReq.setContext(new util.HashMap[String, AnyRef]() {
+      {
+        putAll(request.getContext)
+      }
+    })
+    getCategoryReq.getContext.put(Constants.SCHEMA_NAME, Constants.CATEGORY_INSTANCE_SCHEMA_NAME)
+    getCategoryReq.getContext.put(Constants.VERSION, Constants.CATEGORY_INSTANCE_SCHEMA_VERSION)
+    getCategoryReq.put(Constants.IDENTIFIER, categoryInstanceId)
+    DataNode.read(getCategoryReq) recoverWith {
+      case e: ResourceNotFoundException => {
+        if (StringUtils.equalsAnyIgnoreCase("POST", requestMethod)) {
+          DataNode.read(getCategoryReq)
```

Comment:
What is the `requestMethod`. I'm not sure about this logic. Let us discuss about it.